src/core/pipe/pipe.filter.js

```diff
@@ -0,0 +1,21 @@
+import * as guard from '../infrastructure/guard';
+
+export default function (data, ctx, next) {
+	const model = ctx.model;
+	guard.notUndefined(model.filter, 'model.filter');
+
+	const filter = model.filter();
+	guard.invokable(filter.match, 'model.filter.match');
```

Comment:
i think we don't need to guard it like that, need to think about model type persisting...